kernel/yosys.cc

```diff
@@ -868,7 +868,10 @@ std::string proc_self_dirname()
 	return "/";
 }
 #else
-	#error "Don't know how to determine process executable base path!"
+std::string proc_self_dirname()
+{
+	return PREFIX "/bin";
+}
```

Comment:
Again, some difference here: the ports is specific to OpenBSD (`#ifdef __OpenBSD__`) and use `${PREFIX}` inside the string, I am not sure if it works.

Here is something that could cover more OSes than just OpenBSD.